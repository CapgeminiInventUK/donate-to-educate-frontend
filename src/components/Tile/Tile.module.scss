@use '@assets/global.scss' as g;

.tile {
  background: g.$primary-white;
  padding: 40px;
  border-radius: 10px;
  flex: 0 0 100%;
  border: 0.5px solid grey;
  box-shadow: 0 5px 5px #e5e6eb;
  cursor: pointer;
  flex: 1;

  h2,
  p {
    margin-bottom: 20px;
  }

  h2 {
    margin-top: 20px;
  }
}

.noShadow {
  box-shadow: none;
}

.children {
  margin: 30px 0 10px;
}

@media only screen and (max-width: 1200px) {
  .small {
    max-width: 100%;
  }
}
