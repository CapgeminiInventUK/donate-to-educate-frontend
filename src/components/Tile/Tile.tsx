import { FC } from 'react';
import styles from './Tile.module.scss';
import { motion } from 'framer-motion';

interface TileProps {
  title: string;
  onClick: () => void;
  body: string[];
  icon: JSX.Element;
  size: string;
  children?: React.ReactNode;
  noShadow?: boolean;
}

const Tile: FC<TileProps> = ({ onClick, title, body, icon, size, noShadow, children }) => {
  return (
    <motion.div
      className={`${styles.tile} ${size === 'small' ? styles.small : styles.medium} ${noShadow === true ? styles.noShadow : ''}`}
      whileHover={{ scale: 1.05 }}
      onClick={onClick}
    >
      {icon}
      <h2>{title}</h2>
      {body.map((paragraph) => (
        <p key={paragraph}>{paragraph}</p>
      ))}
      <div className={styles.children}>{children}</div>
    </motion.div>
  );
};

export default Tile;
