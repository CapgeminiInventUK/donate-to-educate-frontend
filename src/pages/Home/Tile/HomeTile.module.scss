@use '@assets/global.scss' as g;

.tile {
  background: g.$primary-white;
  padding: 40px;
  border-radius: 10px;
  max-width: 350px;
  flex: 0 0 100%;

  h2,
  p {
    margin-bottom: 40px;
  }

  h2 {
    margin-top: 40px;
  }
}
