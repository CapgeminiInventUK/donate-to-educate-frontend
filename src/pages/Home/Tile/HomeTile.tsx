import FormButton from '@/components/FormButton/FormButton';
import { motion } from 'framer-motion';
import type { FC } from 'react';
import styles from './HomeTile.module.scss';

interface TileProps {
  title: string;
  onClick: () => void;
  body: string[];
  icon: JSX.Element;
  buttonText: string;
}

const HomeTile: FC<TileProps> = ({ onClick, title, body, icon, buttonText }) => {
  return (
    <motion.div className={styles.tile} whileHover={{ scale: 1.05 }}>
      {icon}
      <h2>{title}</h2>
      {body.map((paragraph) => (
        <p key={paragraph}>{paragraph}</p>
      ))}
      <FormButton
        theme="formButtonGreen"
        text={buttonText}
        onClick={onClick}
        useArrow
        ariaLabel={buttonText}
      />
    </motion.div>
  );
};

export default HomeTile;
