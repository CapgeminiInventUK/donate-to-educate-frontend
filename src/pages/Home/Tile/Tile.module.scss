@use '@assets/global.scss' as g;

.tile {
  background: g.$primary-white;
  padding: 40px;
  border-radius: 10px;
  max-width: 350px;
  flex: 0 0 100%;

  h2,
  p {
    margin-bottom: 40px;
  }

  h2 {
    margin-top: 40px;
  }

  .iconContainer {
    border-radius: 50%;
    width: 50px;
    height: 50px;
    background: #4a8fe7;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
