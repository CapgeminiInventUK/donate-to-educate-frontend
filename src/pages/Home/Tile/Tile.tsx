import { FC } from 'react';
import styles from './Tile.module.scss';
import FormButton from '@/components/FormButton/FormButton';

interface TileProps {
  title: string;
  onClick: () => void;
  body: string[];
  icon: JSX.Element;
  buttonText: string;
}

const Tile: FC<TileProps> = ({ onClick, title, body, icon, buttonText }) => {
  return (
    <div className={styles.tile}>
      {icon}
      <h2>{title}</h2>
      {body.map((paragraph) => (
        <p key={paragraph}>{paragraph}</p>
      ))}
      <FormButton theme="formButtonGreen" text={buttonText} onClick={onClick} useArrow />
    </div>
  );
};

export default Tile;
