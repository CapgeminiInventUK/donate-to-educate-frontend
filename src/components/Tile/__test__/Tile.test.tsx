import { render } from '@testing-library/react';
import Tile from '../Tile';
import userEvent from '@testing-library/user-event';

afterEach(() => {
  vi.resetAllMocks();
});

describe('Tile', () => {
  const props = {
    title: 'Your charity profile is active',
    body: ['View, edit and update your public facing profile.'],
    size: 'medium',
    noShadow: true,
    hoverScale: 1,
    titleLarge: true,
    children: <>children</>,
    icon: <></>,
  };
  it('should render component', () => {
    expect(render(<Tile {...props} />)).toMatchSnapshot();
  });

  it('should handle click', async () => {
    const onClick = vi.fn();
    const { getByLabelText } = render(
      <Tile onClick={onClick} {...props} noShadow={false} size={'small'} titleLarge={false} />
    );

    const tile = getByLabelText('tile');
    await userEvent.click(tile);

    expect(onClick).toHaveBeenCalled();
  });
});
