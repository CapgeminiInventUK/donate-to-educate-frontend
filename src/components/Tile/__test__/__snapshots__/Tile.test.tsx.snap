// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tile > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="tile"
        class="_tile_34c135 undefined _noShadow_34c135    "
      >
        <h2>
          Your charity profile is active
        </h2>
        <p>
          View, edit and update your public facing profile.
        </p>
        <div
          class="_children_34c135"
        >
          children
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="tile"
      class="_tile_34c135 undefined _noShadow_34c135    "
    >
      <h2>
        Your charity profile is active
      </h2>
      <p>
        View, edit and update your public facing profile.
      </p>
      <div
        class="_children_34c135"
      >
        children
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
